### ISSUE TITLE
Tensorflow Hub Compatibility with WebHDFS

### ISSUE THREAD

**Issue by Rmsharks4** [tensorflow/hub#581]
Hi! I'm trying to use: https://tfhub.dev/google/universal-sentence-encoder/4 with WebHDFS (https://github.com/mtth/hdfs) in python, and I can't seem to get it to work. I downloaded the compressed USE file and placed it onto the Hadoop server. But I guess without the WebHDFS Client, hub.Module() cannot access the URL on its own (HTTP Error 400: Bad Request), and it doesn't accept a file reader either. I kind of need the file to be on my Hadoop Server, so I can't unzip it and keep it locally to access it.

Is there another way to do this?

Update: I extracted all the files from the compressed format and placed in a directory on HDFS - and I still get the same error.

**Comment by Rmsharks4** [comment]
Hi! So I couldn't find a solution using the compressed file format, but here's a workaround I used for my problem:

```
import tensorflow as tf
import tensorflow_hub as hub
```
### 1.  LOAD LAYER FROM TENSORFLOW-HUB
```
hub_url = "https://tfhub.dev/google/universal-sentence-encoder-large/5"
embed = hub.KerasLayer(hub_url)
```
### 2. GET WEIGHTS
```
weights = embed.get_weights()
```
### 3. SAVE WEIGHTS
```
import pickle
with open('weights.pkl', 'wb') as file:
    pickle.dump(weights, file)
```
### 4. GET CONFIG
```
config = embed.get_config()
```
### 5. SAVE CONFIG
```
import json
with open('config.json', 'w') as file:
    json.dump(config, file)
```
### 6. SET CONFIG
```
read = hub.KerasLayer(**config)
```
### 7. SET WEIGHTS
```
read.set_weights(weights)
```
Another solution for other servers might be to use the .h5 format for saving, but as of now, HDFS doesnt support the the .h5 format, so I had to dig into this manually. I'm still keeping this issue open in case someone finds the solution for the original problem - using the compressed file on HDFS.

**Comment by gowthamkpr** [comment]
@jhseu Can you PTAL? Thanks!

**Comment by Rmsharks4** [comment]
Update on the issue - the above mentioned solution does not work because it still tries to access the Tensorflow-Hub URL inside the configs - in order to use it with HDFS, I tried this:

```
model = hub.load('http://<DATANODE>:<PORT>/webhdfs/v1/<PATH>?op=OPEN')
```

and it works! Before this, I was not appending ?op=OPEN to the URL - which is why the handler inside tensorflow_hub could not access the URL, but its working now. Closing this issue!
